src/main/java/org/opentestsystem/ap/ims/client/GitlabClient.java

```diff
@@ -0,0 +1,178 @@
+
+import java.io.IOException;
+import javax.annotation.PostConstruct;
+
+import lombok.extern.slf4j.Slf4j;
+import org.gitlab.api.GitlabAPI;
+import org.gitlab.api.http.GitlabHTTPRequestor;
+import org.gitlab.api.models.GitlabGroup;
+import org.gitlab.api.models.GitlabProject;
+import org.gitlab.api.models.GitlabSession;
+import org.opentestsystem.ap.ims.config.ItemBankProperties;
+import org.opentestsystem.ap.ims.util.SystemException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * A client into Gitlab.  The client is configured with a group.  The group
+ * is the top level where all items (or projects) are created and assumed
+ * to live.  For example, looking up a project expects the project to be
+ * under the configured group.
+ */
+@Slf4j
+public class GitlabClient {
+
+    protected final ItemBankProperties gitlabProperties;
+
+    private GitlabSession gitlabSession;
+
+    private GitlabAPI gitlabApi;
+
+    private GitlabGroup gitlabGroup;
+
+    private String gitLabProjectLookupBaseUrl;
+
+    @Autowired
+    public GitlabClient(final ItemBankProperties gitlabProperties) {
+        this.gitlabProperties = gitlabProperties;
+    }
+
+    @PostConstruct
+    public void initialize() {
+        setGitlabSession(createSession());
+
+        setGitlabApi(createGitlabAPI());
+
+        final GitlabGroup gitlabGroup = lookupGroup();
+        setGitlabGroup(gitlabGroup);
+
+        setGitLabProjectLookupBaseUrl(GitlabProject.URL + "/" + gitlabProperties.getGroup() + "%2F");
+    }
+
+    public GitlabProject createProject(String itemId) {
+        final GitlabProject project;
+        try {
+            // using deprecated method, only way to create project under group
+            project = gitlabApi.createProject(
+                (String) itemId,                    // name
+                (Integer) gitlabGroup.getId(),      // namespaceId
+                (String) null,                      // description
+                Boolean.TRUE,                       // issuesEnabled
+                Boolean.FALSE,                      // wallEnabled
+                Boolean.FALSE,                      // mergeRequestsEnabled
+                Boolean.FALSE,                      // wikiEnabled
+                Boolean.FALSE,                      // snippetsEnabled
+                Boolean.FALSE,                      // publik
+                0,                                  // visibilityLevel
+                (String) null                       // importUrl
+            );
+        } catch (IOException e) {
+            throw new SystemException("Error creating Gitlab project", e);
+        }
+        return project;
+    }
+
+    /**
+     * Delete's a project from gitlab.
+     *
+     * @param projectName The project to delete.
+     */
+    public void deleteProject(String projectName) {
+        final GitlabProject gitlabProject = lookupProjectByName(projectName);
+        if (gitlabProject != null) {
```

Comment:
Great point.  Let me get in there and see how that functions.   I will comment more on your other comment.